packages/block-library/src/media-text/index.php

```diff
@@ -29,15 +29,32 @@ function render_block_core_media_text( $attributes, $content ) {
 		return $content;
 	}
 
+	$has_media_on_right = isset( $attributes['mediaPosition'] ) && 'right' === $attributes['mediaPosition'];
```

Comment:
I struggling a bit to follow the code here. In my mind since we now always render the image inside the `figure`, can we simplify the logic?

I would expect for the `img` tag to already be part of the saved content. Isn't that the case?

Reply:
It tripped me up at first as well, but my understanding of it is that the code is accounting for the possibility that there may be multiple `<figure>` elements inside the block, and has different code paths to select the correct one depending on whether the media is on the right, or the left.

In the case where the media is on the right, it chooses the last one. In the case where it's on the left, it chooses the first one.

So while there will always be an `<img>` element in there, as you say, the issue is finding the correct one.

Reply:
Ah, that makes sense since we can have more images in the block's text section..